.github/workflows/cmd.yml

```diff
@@ -0,0 +1,417 @@
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - id: set-image
+        run: |
+          BODY=$(echo "${{ github.event.comment.body }}" | xargs)
+          IMAGE_OVERRIDE=$(echo $BODY | grep -oe 'docker.io/paritytech/ci-unified:.*\s' | xargs)
+          
+          cat .github/env >> $GITHUB_OUTPUT
+          
+          if [ -n "$IMAGE_OVERRIDE" ]; then
+              echo "IMAGE=$IMAGE_OVERRIDE" >> $GITHUB_OUTPUT
+          fi
+          
+          if [[ $BODY == "/cmd bench"* ]]; then
+              echo "RUNNER=arc-runners-polkadot-sdk-benchmark" >> $GITHUB_OUTPUT
+          elif [[ $BODY == "/cmd update-ui"* ]]; then
+              echo "RUNNER=arc-runners-polkadot-sdk-beefy" >> $GITHUB_OUTPUT
+          else
+              echo "RUNNER=ubuntu-latest" >> $GITHUB_OUTPUT
+          fi
+
+  cmd:
+    needs: [ set-image ]
+    env:
+      JOB_NAME: 'cmd'
+    runs-on: ${{ needs.set-image.outputs.RUNNER }}
+    container:
+      image: ${{ needs.set-image.outputs.IMAGE }}
+    steps:
+      - name: Get command
+        uses: actions-ecosystem/action-regex-match@v2
+        id: get-pr-comment
+        with:
+          text: ${{ github.event.comment.body }}
+          regex: '^(\/cmd )([-\/\s\w.=:]+)$' # not ";" or && || to reduce risk of injections
+
+      - name: Build workflow link
+        if: ${{ !contains(github.event.comment.body, '--quiet') }}
+        id: build-link
+        run: |
+          # Get exactly the CMD job link, filtering out the other jobs
+          jobLink=$(curl -s \
+            -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[] | select(.name | contains("${{ env.JOB_NAME }}")) | .html_url')
+          
+          runLink=$(curl -s \
+            -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }} | jq '.html_url')
+          
+          echo "job_url=${jobLink}"
+          echo "run_url=${runLink}"
+          echo "job_url=$jobLink" >> $GITHUB_OUTPUT
+          echo "run_url=$runLink" >> $GITHUB_OUTPUT
+
+
+      - name: Comment PR (Start)
+        if: ${{ !contains(github.event.comment.body, '--quiet') }}
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            let job_url = ${{ steps.build-link.outputs.job_url }}
+            
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: `Command "${{ steps.get-pr-comment.outputs.group2 }}" has started 🚀 [See logs here](${job_url})`
```

Comment:
Yea need to be careful with the injections here.

Reply:
Correct, that's why there's regex above.
Assuming that these commands only for partytech members, that's unlikely to have someone trying to break it intentionally, and unintentional cases we can cover per use-case